client/src/ui/loadouts.ts

```diff
@@ -127,13 +44,15 @@ const loadout = {
     modified: function (a: Loadout, b: Loadout) {
         return !deepEqual(a, b);
     },
-    getUserAvailableItems: function (heroItems: unknown[]) {
-        const items = [];
+    getUserAvailableItems: function (heroItems: Item[] = []) {
+        const processedItemTypes = new Set<string>();
```

Comment:
why is this set necessary
shouldn't duplicated items validation be done server side?